lib/knex/db.ts

```diff
@@ -1306,13 +1306,15 @@ module.exports = function (PersistObjectTemplate) {
                     //When the property type is not an object template, need to compare the values.
                     //for date and object types, need to compare the stringified values.
                     var oldKey = '_ct_org_' + prop;
-                    if (!props[prop].type.isObjectTemplate && (obj[oldKey] !== obj[prop] || ((props[prop].type === Date || props[prop].type === Object) &&
-                        JSON.stringify(obj[oldKey]) !== JSON.stringify(obj[prop]))))  {
+                    if (!props[prop].type.isObjectTemplate && (obj[oldKey] != obj[prop] || ((props[prop].type === Date || props[prop].type === Object) &&
+                        JSON.stringify(obj[oldKey], function (k, v) { return v === null ? undefined : v; }) !== JSON.stringify(obj[prop], function (k, v) { return v === null ? undefined : v; }))))  {
```

Comment:
the block of logic for this if condition seems a little dense - can we separate this out a little so we don't have all of this in one continuous line? e.g.

```const replaceNullValuesWithUndefined = function (k, v) { return v === null ? undefined : v; })``` 

so it might look like...
```JSON.stringify(obj[oldKey], replaceNullValuesWithUndefined) !== JSON.stringify(obj[prop], replaceNullValuesWithUndefined)))```

not necessary for this MR, but going even further we might store the JSON.stringified results in variables like `stringifiedPropOld` and `stringifiedPropNew`, so this entire thing would just read as ``` ... && stringifiedPropOld !== stringifiedPropNew``` this is basically what's described in the // comment, but as is you'd have to parse a lot of code just to figure out that's what it actually does. I'm fine with not making this particular change in the MR, but probably the first one of separating out the filter functions would be good.